### PR TITLE
Task_02 by IgorFromUA

### PR DESCRIPTION
**Pull request by IgorFromUA**
### Acceptance criteria for 1
- [x] Mentor is added to pr
- [x]  AC copied to pr description
- [x]  Link to deployed version is added to pr
- [x]  Interfaces for Product and Category based on [swagger](https://ma-backend-api.mocintra.com/docs#/categories/CategoriesController_getAll) are created and exported in separate file.
- [x]  Static header and footer are created according to [mobile](https://www.figma.com/file/mRAOMdJiAsD8fHSVnYwUl9/React-web-page?type=design&node-id=13-17490&mode=design&t=3Thf45kJJYMoxsc0-0) and [desktop](https://www.figma.com/file/MIKTcyr6SvspkZnIQ5nEPD/MA-Courses-2022-2023?type=design&node-id=2471-2253&mode=design&t=4DJ0BIdFZZ4mUgAn-0) designs.
### Acceptance criteria for 1*
- [x] Shared component SocialIcon is created and used for footer icons
- [x]  Link to social network and path to icon are passed to component by the typed props
### [link to deployed](https://igorfromua.github.io/fe-react-2024/)

### CODE REVIEW

File: src/interfaces/product.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import type Category from './categorie';
+
+interface Product {
+    id: number;
+    title: string;
+    price: number;
+    description: string;
+    images: string[];
+    creationAt: string;
```
Comment (vadym-zinchenko-moc) [inline src/interfaces/product.ts]: it's `Date` type. Pls check it and fix for all interfaces where we have Date.

File: src/components/socialLink/SocialLinks.component.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { socialNetworks } from '../../constants/socialNetworks';
+
+import styles from './socialLink.module.css';
+
+function SocialLinks() {
+    return (
+        <ul className={styles.links}>
+            {socialNetworks.map((network) => (
+                <li key={network.name}>
```
Comment (vadym-zinchenko-moc) [inline src/components/socialLink/SocialLinks.component.tsx]: if you did task 1*. You need to create another component `SocialLink` with own props.

File: src/components/header/header.module.css [hunk 3]
```diff
@@ -1,12 +1,134 @@
...
+  font-family: inherit;
+  font-size: 16px;
+  font-weight: 500;
+  border-radius: 40px;
+  border: none;
+  padding: 5px 10px;
+}
+
+.authButtonLogin {
+  background: #333;
```
Comment (vadym-zinchenko-moc) [inline src/components/header/header.module.css:76]: check color pls from design.

File: src/components/header/header.module.css [hunk 4]
```diff
@@ -1,12 +1,134 @@
...
+}
+
+.authButtons {
+  display: none;
+  justify-content: center;
+  align-items: center;
+  gap: 10px;
+}
+
+.authButtons button {
```
Comment (vadym-zinchenko-moc) [inline src/components/header/header.module.css:60]: why do you use this `.authButtons button `? 
isn't enough class `.authButtons`?

File: src/components/header/Header.component.tsx [hunk 5]
```diff
@@ -1,2 +1,59 @@
...
+import moon from '../../assets/moon-icon.svg';
+import sun from '../../assets/sun-icon.svg';
+import userAdd from '../../assets/userAdd-icon.svg';
+
 import styles from './header.module.css';
-export const HeaderComponent = () => <header className={styles.header}>Header</header>;
+
+export function HeaderComponent() {
+    return (
+        <>
```
Comment (vadym-zinchenko-moc) [inline src/components/header/Header.component.tsx]: extra `Fragment`

File: src/components/footer/Footer.component.tsx [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import SocialLinks from '../socialLink/SocialLinks.component';
+
+import styles from './footer.module.css';
+
+export function FooterComponent() {
+    return (
+        <>
```
Comment (vadym-zinchenko-moc) [inline src/components/footer/Footer.component.tsx]: extra `Fragment`

File: src/components/about/About.component.tsx [hunk 7]
```diff
@@ -5,15 +5,18 @@ import styles from './about.module.css';
 export function AboutComponent() {
     return (
         <>
```
Comment (vadym-zinchenko-moc) [inline src/components/about/About.component.tsx]: extra `Fragment`

File: src/interfaces/categorie.ts [hunk 8]
```diff
@@ -0,0 +1,9 @@
+interface Category {
+    id: number;
+    name: string;
+    image: string;
+    creationAt: string;
+    updatedAt: string;
+}
+
+export default Category;
```
Comment (vadym-zinchenko-moc) [inline src/interfaces/categorie.ts]: better don't use `export default` in future

File: src/components/header/Header.component.tsx [hunk 9]
```diff
@@ -1,2 +1,59 @@
...
+            <header className={styles.header}>
```
Comment (vadym-zinchenko-moc) [inline src/components/header/Header.component.tsx]: i think burger too early to use. 
Doing it on much smaller screens.

Screenshot 2024-04-28 at 22 19 16

**Review by vadym-zinchenko-moc — CHANGES_REQUESTED** [review]

File: src/interfaces/product.ts (hunk 1 above)
Comment (IgorFromUA) [reply src/interfaces/product.ts]: fixed

File: src/components/header/header.module.css (hunk 3 above)
Comment (IgorFromUA) [reply src/components/header/header.module.css:76]: fixed

File: src/components/header/header.module.css (hunk 4 above)
Comment (IgorFromUA) [reply src/components/header/header.module.css:60]: The .authButtons selector is a group of buttons, and the .authButtons button selector is a specifically button

File: src/components/header/Header.component.tsx (hunk 5 above)
Comment (IgorFromUA) [reply src/components/header/Header.component.tsx]: fixed

File: src/components/footer/Footer.component.tsx (hunk 6 above)
Comment (IgorFromUA) [reply src/components/footer/Footer.component.tsx]: fixed

File: src/components/about/About.component.tsx (hunk 7 above)
Comment (IgorFromUA) [reply src/components/about/About.component.tsx]: fixed

File: src/interfaces/categorie.ts (hunk 8 above)
Comment (IgorFromUA) [reply src/interfaces/categorie.ts]: fixed

File: src/components/header/Header.component.tsx (hunk 9 above)
Comment (IgorFromUA) [reply src/components/header/Header.component.tsx]: fixed

File: src/components/socialLink/SocialLinks.component.tsx (hunk 2 above)
Comment (IgorFromUA) [reply src/components/socialLink/SocialLinks.component.tsx]: Done

**Review by vadym-zinchenko-moc — APPROVED** [review]

**Pull request merged**